indigo_app/views/works.py

```diff
@@ -82,6 +82,29 @@ def get_work_timeline(self, work):
         # add expressions
         for entry in timeline:
             entry.expressions = [e for e in work_expressions if e.expression_date == entry.date]
+        # add tasks
+        timeline_tasks = work.tasks.filter(timeline_date__isnull=False).exclude(state=Task.CANCELLED)
+        dates = [entry.date for entry in timeline]
+        # simple case: add tasks to existing corresponding entries
+        for entry in timeline:
+            entry_tasks = timeline_tasks.filter(timeline_date=entry.date)
+            entry.tasks = list(entry_tasks)
```

Comment:
only when `list()` is called will it go to the db; if there are 50 items in the timeline, then it's going go to 50 times.

Instead, pull it into the list on line 86, then filter on line 90 entirely in python, eg `[t for t in timeline_tasks if t.timeline_date == entry.date]`

Reply:
gotcha!

Reply:
I made a separate list for the simple tasks, as the extra tasks still need to filter timeline_tasks to get started.